src/v/bytes/iobuf.h

```diff
@@ -359,11 +361,10 @@ iobuf::append(const uint8_t* src, size_t len) {
     }
 }
 
-/// appends the contents of buffer; might pack values into existing space
-[[gnu::always_inline]] inline void iobuf::append(ss::temporary_buffer<char> b) {
-    if (unlikely(!b.size())) {
-        return;
-    }
+// tries to pack the contents of the buffer into the last fragment of an iobuf
```

Comment:
This isn't correct? It doesn't just back it into the fragment (it doesn't examine the remaining space at all), but mem-copies it into the iobuf if it passes the heuristic (though I guess that's usually at most across 2 fragments).